src/backend/cdb/cdbpath.c

```diff
@@ -1235,7 +1236,15 @@ cdbpath_motion_for_join(PlannerInfo *root,
 	 */
 	else if (CdbPathLocus_IsReplicated(outer.locus))
 	{
-		if (root->upd_del_replicated_table > 0)
+		if (root->parse->hasModifyingCTE && root->parse->commandType == CMD_SELECT)
+		{
+			/*
+			 * ignore special cases for SELECT over DML, we don't need to make
+			 * inner part of join replicated and add additional motion layers
+			 * later
+			 */
```

Comment:
As far I understand, we should handle `CdbLocusType_Replicated` in these two blocks. There are asserts for only Bottlenecks and Partitioned locuses in blocks below.
But we never add motions in case of `CdbLocusType_Replicated` type in the current implementation. When is it a problem?